scripts/check-pipeline.py

```diff
@@ -0,0 +1,83 @@
+import os
+import sys
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any
+
+import requests
+from dateutil import parser
+
+
+@dataclass
+class _WorkflowStatus:
+    created_at: datetime
+    name: str
+    status: str
+
+
+_PIPELINE_URL = "https://circleci.com/api/v2/pipeline/{pipeline_id}/workflow"
+_WORKFLOW_URL = "https://circleci.com/api/v2/workflow/{workflow_id}"
+_STATUSES: dict[str, _WorkflowStatus] = {}
+
+
+def _process_data(items: list[dict[str, Any]]) -> None:
+    for workflow in items:
+        created_at = parser.parse(workflow["created_at"])
+        name = workflow["name"]
+        status = workflow["status"]
+        if name not in _STATUSES:
+            _STATUSES[name] = _WorkflowStatus(created_at, name, status)
+        elif _STATUSES[name].created_at == created_at and _STATUSES[name].status != status:
+            _STATUSES[name].status = status
+        elif _STATUSES[name].created_at < created_at:
+            status[name] = _WorkflowStatus(created_at, name, status)
+
+
+def main() -> None:
+    api_token = os.environ["API_TOKEN"]
+    workflow_id = sys.argv[1]
+    headers = {
+        "Circle-Token": api_token,
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+    }
+    page_token = None
+    url = _WORKFLOW_URL.format(workflow_id=workflow_id)
+    res = requests.get(url, headers=headers)
+    data = res.json()
+    pipeline_id = data["pipeline_id"]
```

Comment:
Can we not get the `pipeline_id` directly via `sys.argv`, instead of `workflow_id`?

Reply:
basically no, https://circleci.com/docs/variables/#built-in-environment-variables